### ISSUE TITLE
Service Discovery doesn't return an error when an incorrect password is provided.

### ISSUE THREAD

**Issue by andysCaplin** [Refinitiv/Real-Time-SDK#194]
We have an ETA application that uses Discovery to find the the list of endpoints we're permissioned to test against.

We call rsslReactorQueryServiceDiscovery() to discover the endpoints and it discovers them fine when we provide the correct credentials. 

But as part of our testing we provided an incorrect password to the RsslReactorServiceDiscoveryOptions that we pass to rsslReactorQueryServiceDiscovery(). The routine returns RSSL_RET_SUCCESS in this instance - which we think is incorrect.

The same thing happens if we provide an incorrect clientId or userName in RsslReactorServiceDiscoveryOptions.

We would expect in this case that the routine would return a failure code and that the RsslErrorInfo would detail the error.

The version we're using is
```
PACKAGE eta3.6.4.L1 rrg  VERSION eta3.6.4.0  Tue Mar 15 22:07:55 CST 2022
```

We have a mode where we can skip Discovery and use a specific endpoint directly. In that case we get an expected failure code and error message when we attempt to connect
```
Failed to request authentication token information with HTTP error 400. Text: {"error":"access_denied"  ,"error_description":"Account locked out due to provide incorrect password."
```

**Comment by fogol** [comment]
@andysCaplin Thank you for bringing this to our attention. We will investigate the issue. It will be fixed in further releases.

**Comment by fogol** [comment]
@andysCaplin This is an error in the application (not the library). You can make a simple change in this manner: change the application to check RsslReactorServiceEndpointEvent.pErrorInfo to see whether there is an error and react accordingly. Here's a code sample for how to do this:

in this function in ETAC Watchlist Consumer code,

`RsslReactorCallbackRet serviceEndpointEventCallback(RsslReactor *pReactor, RsslReactorServiceEndpointEvent *pEndPointEvent)`

, there needs to be a check like this:

`	if (pEndPointEvent->pErrorInfo)
	{
		printf("Retrieve endpoints failed. Text: %s\n", pEndPointEvent->pErrorInfo->rsslError.text);
	}
`

**Comment by andysCaplin** [comment]
You mean that that if rsslReactorQueryServiceDiscovery() fails in this way it returns SUCCESS and it's not sufficient to check that code but we need to look into pEndPointEvent ?

The example I found that uses that API doesn't make that clear
./Cpp-C/Eta/Applications/Examples/WatchlistConsumer/rsslWatchlistConsumer.c
```                if(rsslReactorQueryServiceDiscovery(pReactor, &serviceDiscoveryOpts, &rsslErrorInfo) != RSSL_RET_SUCCESS)
                {
                        printf("Error: %s\n", rsslErrorInfo.rsslError.text);
                        exit(-1);
                }
```
Also there's nothing in the SDK for the API that indicates that we should look in the error text if SUCCESS is returned.

If it is the case that the API can fail and return SUCCESS shouldn't the documentation be updated ?

**Comment by fogol** [comment]
@andysCaplin Thank you for you response! Yes, the example will be updated to fix this issue.

> Also I can see in ./Cpp-C/Eta/Impl/Reactor/rsslReactor.c whenever pOpts->pServiceEndpointEventCallback() is called its return code is ignored. If it fails shouldn't that failure be returned to the caller of rsslReactorQueryServiceDiscovery() ?

It is by design now. The idea is to use rsslReactorQueryServiceDiscovery() to deal with REST request and response and to provide the result within ServiceEndpointEventCallback() call. So rsslReactorQueryServiceDiscovery() has finished successfully and provided the result in callback. We will discuss this design if it needs to be changed.

The example will be fixed in further releases. Thank you for this update!

**Comment by andysCaplin** [comment]
Ok - it does seem then that the callback should be a void method - because the return code is not used. If it's not void it gives the impression that the return code is used in some way.
Or else - maybe just doc in the example that return code is unused and should just be set to SUCCESS ?

**Comment by vlevendel** [comment]
Available with tag Real-Time-SDK-2.0.7.L1
